torch/distributed/_tensor/_dispatch.py

```diff
@@ -374,7 +384,6 @@ def try_get_replicate_spec(
             else:
                 kwargs_schema[k] = v
                 local_kwargs[k] = v
-
```

Comment:
before landing this, I probably need to:

(1) try adding a sharding prop rule for `aten._foreach_mul`
(2) see if that fixes the repro and add tests for it